legacy-commands-v1/src/main/java/io/github/minecraftcursedlegacy/impl/command/DefaultCommandDispatcherImpl.java

```diff
@@ -23,48 +23,49 @@
 
 package io.github.minecraftcursedlegacy.impl.command;
 
-import io.github.minecraftcursedlegacy.api.command.CommandDispatchEvent;
-import io.github.minecraftcursedlegacy.api.command.DefaultCommandDispatcher;
-import net.fabricmc.api.EnvType;
-
-import javax.annotation.Nullable;
 import java.util.HashMap;
 import java.util.Map;
 
+import javax.annotation.Nullable;
+
+import io.github.minecraftcursedlegacy.api.command.DefaultCommandDispatcher;
+import io.github.minecraftcursedlegacy.api.command.DispatcherRegistry;
+import io.github.minecraftcursedlegacy.api.command.Sender;
+import net.fabricmc.api.EnvType;
+
+/**
+ * The implementation for the default command dispatcher.
+ * @since 1.1.0
+ */
 public class DefaultCommandDispatcherImpl implements DefaultCommandDispatcher{
 	private final Map<String, Command> singleplayerCommands = new HashMap<>();
 	private final Map<String, Command> multiplayerCommands = new HashMap<>();
 
-	public DefaultCommandDispatcherImpl() {
-		CommandDispatchEvent.MULTIPLAYER_DISPATCH.register((source, commandString) -> {
-			String[] args = commandString.split(" ");
-			Command command = multiplayerCommands.get(args[0]);
-			if (command != null) {
-				command.execute(source, args);
-				return true;
-			}
-			return false;
-		});
-
-		CommandDispatchEvent.SINGLEPLAYER_DISPATCH.register((source, commandString) -> {
-			String[] args = commandString.split(" ");
-			Command command = singleplayerCommands.get(args[0]);
-			if (command != null) {
-				command.execute(source, args);
-				return true;
-			}
-			return false;
-		});
-	}
-
 	public void register(String commandName, Command commandFunction, @Nullable EnvType environment) {
-		if (environment == EnvType.SERVER) {
+		if (environment != EnvType.CLIENT) {
 			multiplayerCommands.put(commandName, commandFunction);
-		} else if (environment == EnvType.CLIENT) {
-			singleplayerCommands.put(commandName, commandFunction);
-		} else {
+			DispatcherRegistry.registerSingleplayerDispatcher(commandName, this);
```

Comment:
Should this not be `registerMultiplayerDispatcher`? As the `environment` is anything but the client

Reply:
oh of course